### PR TITLE
Add Extented Log-Likelihood function and unit test

### PR DESCRIPTION
**Pull request by arushanova**
Binned Extended Log-Likelihood function has been added to test_statistic.py
The unit test has been added to test_test_statistic.py

### CODE REVIEW

File: echidna/fit/test_statistic.py [hunk 1]
```diff
@@ -503,3 +503,92 @@ def get_penalty_term(self, current_value, prior, sigma):
...
+        """
+        # Convert to arrays, if observed and expected are floats
+        if isinstance(observed, float):
+            observed = numpy.array([observed])
+        if isinstance(expected, float):
+            expected = numpy.array([expected])
+        # Chosen due to backgrounds with low rates in ROI
+        epsilon = 1e-34  # In the limit of zero
+        total = 0
+        for i in range(len(observed)):
```
Comment (mjmottram) [inline echidna/fit/test_statistic.py]: You can make this more efficient with `for exp, obs in zip(expected, observed)`.  You can make it even more efficient by just doing `expected[expected < epsilon] = epsilon; bin_value = observed * numpy.log(expected)`

Comment (mjmottram) [inline echidna/fit/test_statistic.py]: FYI, the other methods have already been updated to use `zip`, see e.g. https://github.com/snoplusuk/echidna/blob/master/echidna/fit/test_statistic.py#L167.

File: echidna/test/test_test_statistic.py [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import numpy
 
 from echidna.fit.test_statistic import (TestStatistic, BakerCousinsChi,
-                                        BakerCousinsLL, Neyman, Pearson)
+                                        BakerCousinsLL, Neyman, Pearson, ExtendedLL)
```
Comment (mjmottram) [inline echidna/test/test_test_statistic.py]: Breaks pep8iness :)

File: echidna/test/test_test_statistic.py [hunk 3]
```diff
@@ -146,3 +146,29 @@ class that calculates the poisson likelihood chi-squared.
...
+        test_statistic_pb = ExtendedLL()
+        result = test_statistic.compute_statistic(self._observed,
+                                                  self._expected)
+        self.assertIsInstance(result, float)
+        self.assertAlmostEquals((result - (-20.8948597368)), 0.)
+        result = test_statistic_pb.compute_statistic(self._observed,
+                                                     self._expected)
+        calculated = numpy.array([-10.5333264925, -1.58676627531, 1.0,
+                                   0.1, 1.56575786324, 1.0,
+                                   -1.48972567003, -10.9507991622])
```
Comment (mjmottram) [inline echidna/test/test_test_statistic.py]: ditto lines 171 and 172; indents should have one space fewer.

**Comment by mjmottram** [comment]
To be consistent with the way PRs have been managed in the past, I think pep8ing code including the tests makes sense.

**Comment by ashleyrback** [comment]
Yes, tests should be pep8 compliant as well. Really any python code in the `echidna` directory should be pep8 checked.

File: echidna/test/test_test_statistic.py (hunk 3 above)
Comment (arushanova) [inline echidna/test/test_test_statistic.py]: I'll change the settings. I pep8 checked, it didn't break!

**Pull request merged**